### PR TITLE
Add factorio-map-process.sh script

### PR DESCRIPTION
**Pull request by NateMeyer**
This script can be used to update a webserver hosting datestamped world maps.

### CODE REVIEW

File: factorio-map-process.sh [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        -s|--server)
+            SERVER_BASE_PATH="$2"
+            echo "Using Server Path $SERVER_BASE_PATH"
+            shift
+            shift
+            ;;
+        *)
+            TARFILE=$1
+            echo "TARFILE $TARFILE"
+            if [[ $TARFILE != *.tar ]]; then
```
Comment (Rycieos) [inline factorio-map-process.sh]: The python program can read from any tar file, compressed or not. This will only match uncompressed (by name at least) tar files.

File: factorio-map-process.sh [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	            echo "ERROR: File is not a .tar file"
+                usage;
+	            exit 0
+            fi
+            FULLNAME=$(basename "$TARFILE")
```
Comment (Rycieos) [inline factorio-map-process.sh]: `FULLNAME` is a weird var name when it's only the filename. Maybe you meant `FILENAME`?

File: factorio-map-process.sh [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            ;;
+        -r)
+            REDUCE=1
+            shift
+            ;;
+        -t|--tiles)
+            MAP_TILES_PATH="$2"
+            echo "Using MAP Path $MAP_TILES_PATH"
+            shift
+            shift
```
Comment (Rycieos) [inline factorio-map-process.sh]: You can call `shift` with arguments.
```suggestion
            shift 2
```

File: factorio-map-process.sh [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+    -s | --server <server path>     Set the base path where the http server files will be stored.  Default value is
+                                    '/srv/factorio/maps'
+
+    -r                              Flag to size-reduce the images folder with the rdfind tool
+EOF
+}
+
+function testRequirements() {
+	if ! [ -x "$(command -v jq)" ]; then
```
Comment (Rycieos) [inline factorio-map-process.sh:35]: This is more complicated that simply using `if` to test the return code:
```suggestion
	if ! command -v jq; then
```

File: factorio-map-process.sh [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+		echo "ERROR: Requirement not satisfied - jq -"
+		exit 1
+	fi
+
+	if ! [ -x "$(command -v python3)" ]; then
+		echo "ERROR: Requirement not satisfied - python3 -"
```
Comment (Rycieos) [inline factorio-map-process.sh]: This test seems a little pointless, since you are testing for `python3`, but not the other dependencies of `factoriomap.py`.

File: factorio-map-process.sh [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+		exit 1
+	fi
+
+}
+
+function parseFileName() {
+	NAME_NOEXT="${FULLNAME%.*}"
+	echo "Target File: $NAME_NOEXT"
+
+	infos=($(awk -F'[_.]' '{print $1; print $2; print $3; print $4}' <<< $NAME_NOEXT))
```
Comment (Rycieos) [inline factorio-map-process.sh]: The filename string format that this script is expecting is documented as `{WorldName}_YYYY_MM_DD.tar`. Why break the  ISO-8601 standard of `YYYY-MM-DD`, using dashes? It would make your parsing much easier; you only have to split once.

File: factorio-map-process.sh [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    fi
+}
+
+function processTiles() {
+    echo "DESTINATION $DESTPATH"
+    mkdir $DESTPATH
+    python3 $FACTORIO_MAP_SCRIPT $TARFILE $DESTPATH
+
+    if [ $REDUCE -eq 1 ]; then
+        rdfind -makehardlinks true $SERVER_BASE_PATH/$WORLDNAME/$MAP_TILES_PATH
```
Comment (Rycieos) [inline factorio-map-process.sh]: What if I don't want to add an extra layer here for `$WORLDNAME`? I don't use that right now, I have my root as `/var/factoriomaps/`, and the tiles path as `images/`. I'd like to see a way to support not having that, ideally as it parsing the filename, and on not seeing a `_`, it sets `$WORLDNAME` to blank.

File: factorio-map-process.sh [hunk 8]
```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+## Defaults
+SERVER_BASE_PATH=/srv/factorio/maps
+MAP_TILES_PATH=images
+FACTORIO_MAP_SCRIPT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/factoriomap.py
```
Comment (Rycieos) [inline factorio-map-process.sh]: This looks brittle. It is using `&&`, so if the `cd` fails, you won't try to print the working directory, which I guess is good. But then this var will resolve to `/factoriomap.py`, which is for sure not right.

You should set `set -euo pipefail` as the second line of this script, which will help some. See: https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/. These are accepted best practice.

I think the extra `cd && pwd` step isn't needed. I would go with
```suggestion
FACTORIO_MAP_SCRIPT="$(dirname "${BASH_SOURCE[0]}")/factoriomap.py"
```
I could be wrong though, I know that "finding the source directory of the script I'm running" is a very hard problem. I've struggled with it before.

File: factorio-map-process.sh [hunk 9]
```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
```
Comment (Rycieos) [inline factorio-map-process.sh]: Thoughts on this instead?
```suggestion
#!/usr/bin/env bash
```

File: factorio-map-process.sh [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+## Defaults
```
Comment (Rycieos) [inline factorio-map-process.sh:3]: Is `Defaults` the right word here, when not all of these can be overridden? Not sure if they should be though...

**Review by Rycieos — CHANGES_REQUESTED** [review]

File: factorio-map-process.sh [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+            MAP_TILES_PATH="$2"
+            echo "Using MAP Path $MAP_TILES_PATH"
+            shift
+            shift
+            ;;
+        -s|--server)
+            SERVER_BASE_PATH="$2"
+            echo "Using Server Path $SERVER_BASE_PATH"
+            shift
+            shift
```
Comment (NateMeyer) [inline factorio-map-process.sh]: Also change this to `shift 2`
```suggestion
            shift 2
```

File: factorio-map-process.sh (hunk 1 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: Since tar will automatically figure out if it is compressed, we can use tar to test the archive.
```suggestion
            if  ! { tar tf "$TARFILE"; } >/dev/null 2>&1;; then
```

File: factorio-map-process.sh [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+            ;;
+        *)
+            TARFILE=$1
+            echo "TARFILE $TARFILE"
+            if [[ $TARFILE != *.tar ]]; then
+	            echo "ERROR: File is not a .tar file"
```
Comment (NateMeyer) [inline factorio-map-process.sh]: Update error message describing tar
```suggestion
	            echo "ERROR: File is not a tar archive```

File: factorio-map-process.sh (hunk 2 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: Sure, can change var to `FILENAME`
```suggestion
            FILENAME=$(basename "$TARFILE")
```

File: factorio-map-process.sh [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+
+        if [[ REDUCE -eq 1 && !(-x "$(command -v rdfind)") ]]; then
+		echo "ERROR: Requirement not satisfied - rdfind -"
+		exit 1
+	fi
+
+}
+
+function parseFileName() {
+	NAME_NOEXT="${FULLNAME%.*}"
```
Comment (NateMeyer) [inline factorio-map-process.sh]: Change matching var to `FILENAME`
```suggestion
	NAME_NOEXT="${FILENAME%.*}"
```

File: factorio-map-process.sh (hunk 6 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: The underscores are admittedly personal taste.  I can defer to a standard where it makes sense, like here, especially since I was converting it to that standard in the next section anyways.

Comment (NateMeyer) [reply factorio-map-process.sh]:
```suggestion
	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< $NAME_NOEXT))
```

File: factorio-map-process.sh [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+	echo "Target File: $NAME_NOEXT"
+
+	infos=($(awk -F'[_.]' '{print $1; print $2; print $3; print $4}' <<< $NAME_NOEXT))
+
+	WORLDNAME=${infos[0]}
+	echo "WORLD NAME $WORLDNAME"
+
+	DATESTR=${infos[1]}-${infos[2]}-${infos[3]}
```
Comment (NateMeyer) [inline factorio-map-process.sh]: Update for expecting tar filename to have the date using the standard format.
```suggestion
	DATESTR=${infos[1]}
```

File: factorio-map-process.sh [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+    cat <<-EOF
+USAGE
+
+factorio-map-process.sh [OPTIONS] tarfile
+
+DESCRIPTION
+
+This script will process a given tarfile into a set of images used in a leaflet map.
+The default behavior will create a directory for the world maps if one does not exist and copy the example html page in as 'index.html'
+The tar file will be parsed for a map name and a date, separated by underscores (_). Files should be of the format {WorldName}_YYYY_MM_DD.tar.
```
Comment (NateMeyer) [inline factorio-map-process.sh]: Update description to reflect expecting date in the standard format.

```suggestion
The tar file will be parsed for a map name and a date, separated by underscores (_). Files should be of the format {WorldName}_YYYY-MM-DD.tar.
```

File: factorio-map-process.sh (hunk 10 above)
Comment (NateMeyer) [reply factorio-map-process.sh:3]: I think defaults is fine. All of these are configurable or derived.  If you want to make the path to the python script configurable, that could be a future improvement.

File: factorio-map-process.sh (hunk 4 above)
Comment (NateMeyer) [reply factorio-map-process.sh:35]: This simplification prints out  `/usr/bin/jq` unnecessarily.  The existing test works without this side effect.

File: factorio-map-process.sh (hunk 5 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: I think testing for python3 is fair, it's a top-level check and doesn't assume much about the user's environment.  Also, the dependencies of the python script should be tested by the python script, otherwise updates to the python would need to be also updated here.

File: factorio-map-process.sh [hunk 16]
```diff
@@ -0,0 +1,149 @@
...
+        -s|--server)
+            SERVER_BASE_PATH="$2"
+            echo "Using Server Path $SERVER_BASE_PATH"
+            shift 2
+            ;;
+        *)
+            TARFILE=$1
+            echo "TARFILE $TARFILE"
+            if  ! { tar tf "$TARFILE"; } >/dev/null 2>&1;; then
+	            echo "ERROR: File is not a tar archive```
```
Comment (NateMeyer) [inline factorio-map-process.sh]: Fix typos from first update.
```suggestion
            if  ! { tar tf "$TARFILE"; } >/dev/null 2>&1; then
	            echo "ERROR: File is not a tar archive"
```

File: factorio-map-process.sh (hunk 5 above)
Comment (Rycieos) [reply factorio-map-process.sh]: I'm still not sold. I agree we shouldn't test for the script dependencies here, but I don't think there is much point in testing for python3, since to install this project you need to have it.

File: factorio-map-process.sh [hunk 17]
```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
```
Comment (Rycieos) [inline factorio-map-process.sh]: As I think I mentioned on some other line, I'd like to see
```suggestion
set -euo pipefail
```
as bash options on this script.

Some people like to have it as my suggestion on line 2, others like to have it as part of the shbang line, like this:
```bash
#!/usr/bin/env bash -euo pipefail
```
I prefer the first option.

Blog post about this:
https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/

The reasoning is that Shell scripting tends to act like a bunch of commands, not a programing language, but once you start to get long scripts like this, you tend to need programing language error handling features. Those 3 (error exit, error unset, error pipe fail) are good defaults.

File: factorio-map-process.sh [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+## Defaults
+SERVER_BASE_PATH=/srv/factorio/maps
+MAP_TILES_PATH=images
+FACTORIO_MAP_SCRIPT="$(dirname "${BASH_SOURCE[0]}")/factoriomap.py"
+TARFILE=
+REDUCE=0
+
+function usage() {
+    cat <<-EOF
```
Comment (Rycieos) [inline factorio-map-process.sh]: I learned something today, so thanks.
The `-` in the `<<-EOF` string means something:

https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Here-Documents
> If the redirection operator is ‘<<-’, then all leading tab characters are stripped from input lines and the line containing delimiter. This allows here-documents within shell scripts to be indented in a natural fashion.

This doesn't seem to apply here, since you aren't using tab characters at all (it ignores space chars), and you haven't indented in this string. I guess remove the `-`.
```suggestion
    cat <<EOF
```

File: factorio-map-process.sh [hunk 19]
```diff
@@ -0,0 +1,148 @@
...
+        if [[ REDUCE -eq 1 && !(-x "$(command -v rdfind)") ]]; then
+		echo "ERROR: Requirement not satisfied - rdfind -"
+		exit 1
+	fi
+
+}
+
+function parseFileName() {
+	NAME_NOEXT="${FILENAME%.*}"
+	echo "Target File: $NAME_NOEXT"
```
Comment (Rycieos) [inline factorio-map-process.sh]: Why print the name of the file without its extension?
```suggestion
	echo "Target File: ${FILENAME}"
```

File: factorio-map-process.sh [hunk 20]
```diff
@@ -0,0 +1,148 @@
...
+    if [ $REDUCE -eq 1 ]; then
+        rdfind -makehardlinks true $SERVER_BASE_PATH/$WORLDNAME/$MAP_TILES_PATH
+    fi
+}
+
+function addDatesJSON() {
+    DATEFILE=$SERVER_BASE_PATH/$WORLDNAME/dates.json
+    if [[ -f $DATEFILE ]]; then
+        TMPFILE=$(mktemp)
+        jq --arg date "$DATESTR" '. |= .+ [$date]' $DATEFILE > "$TMPFILE" && cat $TMPFILE > $DATEFILE
```
Comment (Rycieos) [inline factorio-map-process.sh]: A much simpler `jq` statement is
```suggestion
        jq ". += [\"${DATESTR}\"]" "${DATEFILE}" > "${TMPFILE}" && cat "${TMPFILE}" > "${DATEFILE}"
```
As you can see, I also prefer the `${VAR}` format for variables always, and you should _always_ `"` double quote vars that could contain spaces, like paths.

File: factorio-map-process.sh [hunk 21]
```diff
@@ -0,0 +1,148 @@
...
+        rm $TMPFILE
+    else
+        echo -e '[\n    "'$DATESTR'" \n]' >> $DATEFILE
```
Comment (Rycieos) [inline factorio-map-process.sh]: I'd like to use `jq` here too, and it should make this more readable
```suggestion
        jq <<< "[\"${DATESTR}\"]" > "${DATEFILE}"
```
Might as well also use `>` redirection, since you already tested and know that the file doesn't exist. If somehow it did, postpending this string to it would break the JSON format.

File: factorio-map-process.sh [hunk 22]
```diff
@@ -0,0 +1,148 @@
...
+
+    if [ $REDUCE -eq 1 ]; then
+        rdfind -makehardlinks true $SERVER_BASE_PATH/$WORLDNAME/$MAP_TILES_PATH
+    fi
+}
+
+function addDatesJSON() {
+    DATEFILE=$SERVER_BASE_PATH/$WORLDNAME/dates.json
+    if [[ -f $DATEFILE ]]; then
+        TMPFILE=$(mktemp)
```
Comment (Rycieos) [inline factorio-map-process.sh]: always " double quote vars that could contain spaces, like paths.
```suggestion
        TMPFILE="$(mktemp)"
```

File: factorio-map-process.sh [hunk 23]
```diff
@@ -0,0 +1,148 @@
...
+        jq --arg date "$DATESTR" '. |= .+ [$date]' $DATEFILE > "$TMPFILE" && cat $TMPFILE > $DATEFILE
+        rm $TMPFILE
```
Comment (Rycieos) [inline factorio-map-process.sh]: always " double quote vars that could contain spaces, like paths.
```suggestion
        rm "${TMPFILE}"
```

File: factorio-map-process.sh [hunk 24]
```diff
@@ -0,0 +1,148 @@
...
+    fi
+}
+
+function processTiles() {
+    echo "DESTINATION $DESTPATH"
+    mkdir $DESTPATH
+    python3 $FACTORIO_MAP_SCRIPT $TARFILE $DESTPATH
+
+    if [ $REDUCE -eq 1 ]; then
+        rdfind -makehardlinks true $SERVER_BASE_PATH/$WORLDNAME/$MAP_TILES_PATH
```
Comment (Rycieos) [inline factorio-map-process.sh]: always " double quote vars that could contain spaces, like paths.
```suggestion
        rdfind -makehardlinks true "${SERVER_BASE_PATH}/${WORLDNAME}/${MAP_TILES_PATH}"
```

File: factorio-map-process.sh [hunk 25]
```diff
@@ -0,0 +1,148 @@
...
+    fi
+}
+
+function addDatesJSON() {
+    DATEFILE=$SERVER_BASE_PATH/$WORLDNAME/dates.json
```
Comment (Rycieos) [inline factorio-map-process.sh]: always " double quote vars that could contain spaces, like paths.
```suggestion
    DATEFILE="${SERVER_BASE_PATH}/${WORLDNAME}/dates.json"
```

File: factorio-map-process.sh [hunk 26]
```diff
@@ -0,0 +1,148 @@
...
+        ## Create New world folder
+        mkdir -p $SERVER_BASE_PATH/$WORLDNAME/$MAP_TILES_PATH
+        cp example.html $SERVER_BASE_PATH/$WORLDNAME/index.html
+    fi
+}
+
+function processTiles() {
+    echo "DESTINATION $DESTPATH"
+    mkdir $DESTPATH
+    python3 $FACTORIO_MAP_SCRIPT $TARFILE $DESTPATH
```
Comment (Rycieos) [inline factorio-map-process.sh]: What if you changed this to 
```suggestion
    $FACTORIO_MAP_SCRIPT $TARFILE $DESTPATH
```
Then you don't technically need to check for `python3` existing.

File: factorio-map-process.sh (hunk 25 above)
Comment (Rycieos) [reply factorio-map-process.sh]: @NateMeyer if you don't want to go around the whole file fixing quoting and `{}` on vars, I can fix that as part of my merge.

**Review by Rycieos — CHANGES_REQUESTED** [review]

File: factorio-map-process.sh [hunk 27]
```diff
@@ -0,0 +1,148 @@
...
+	if ! [ -x "$(command -v jq)" ]; then
+		echo "ERROR: Requirement not satisfied - jq -"
+		exit 1
+	fi
+
+	if ! [ -x "$(command -v python3)" ]; then
+		echo "ERROR: Requirement not satisfied - python3 -"
+		exit 1
+	fi
+
```
Comment (NateMeyer) [inline factorio-map-process.sh]:
```suggestion
```
Removing Python 3 check

File: factorio-map-process.sh (hunk 20 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: I'll test this out, this does look cleaner.

File: factorio-map-process.sh (hunk 21 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: I'll test this works and change to use `jq`.

File: factorio-map-process.sh (hunk 20 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: Seems to work.

File: factorio-map-process.sh (hunk 21 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: This is good too.

File: factorio-map-process.sh [hunk 28]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+function parseFileName() {
+	NAME_NOEXT="${FILENAME%.*}"
+	echo "Target File: ${FILENAME}"
+
+	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< $NAME_NOEXT))
```
Comment (Rycieos) [inline factorio-map-process.sh]: I tested it, this breaks if the filename has a space in it:
```bash
NAME_NOEXT="my world_2020-01-01"
infos=($(awk -F'[_.]' '{print $1; print $2}' <<< $NAME_NOEXT))
for i in $infos; do echo $i; done
my
world
2020-01-01
```
The `var=( some things )` form uses whitespace splitting to decide where to break into array values. If you use the default IFS var, it will separate the words around the space and the newlines printed by `awk`.
This should fix it:
```suggestion
	IFS='\n'
	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< "${NAME_NOEXT}"))
	unset IFS
```

See https://www.gnu.org/software/bash/manual/html_node/Word-Splitting.html

File: factorio-map-process.sh [hunk 29]
```diff
@@ -0,0 +1,143 @@
...
+            ;;
+        -s|--server)
+            SERVER_BASE_PATH="$2"
+            echo "Using Server Path $SERVER_BASE_PATH"
+            shift 2
+            ;;
+        *)
+            TARFILE=$1
+            echo "TARFILE $TARFILE"
+            if  ! { tar tf "$TARFILE"; } >/dev/null 2>&1; then
```
Comment (Rycieos) [inline factorio-map-process.sh]: I tested it, the `{}` aren't needed. Why did you add them?
```suggestion
            if ! tar tf "$TARFILE" >/dev/null 2>&1; then
```

File: factorio-map-process.sh [hunk 30]
```diff
@@ -0,0 +1,143 @@
...
+	NAME_NOEXT="${FILENAME%.*}"
+	echo "Target File: ${FILENAME}"
+
+	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< $NAME_NOEXT))
+
+	WORLDNAME=${infos[0]}
+	echo "WORLD NAME $WORLDNAME"
+
+	DATESTR=${infos[1]}
+	echo "DATE STRING $DATESTR"
```
Comment (Rycieos) [inline factorio-map-process.sh]: I would still like to be able to support only one world. That is, not having a world directory under my base path.

Maybe something like this could be done here?
```suggestion
	if [[ ${#infos[@]} -ge 2 ]]; then
		WORLDNAME="${infos[0]}"
		echo "WORLD NAME $WORLDNAME"

		DATESTR=${infos[1]}
		echo "DATE STRING $DATESTR"
	else
		WORLDNAME=""
		DATESTR=${infos[0]}
		echo "DATE STRING $DATESTR"
	fi
```
By setting `WORLDNAME` to an empty string, we are setting it, which will prevent errors with the `set -u` at the start of the script. But being empty, when we use it to create a path, that dir will be skipped.

**Review by Rycieos — CHANGES_REQUESTED** [review]

File: factorio-map-process.sh (hunk 30 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: I like this solution.

File: factorio-map-process.sh (hunk 4 above)
Comment (Rycieos) [reply factorio-map-process.sh:35]: Looking at it again, you are doing a double check here. You use `command` to fine a binary that matches that name, then `-x` to check if that binary is executable. This would be simpler:
```suggestion
	if ! command -v jq >/dev/null; then
```

File: factorio-map-process.sh [hunk 31]
```diff
@@ -0,0 +1,151 @@
...
+EOF
+}
+
+function testRequirements() {
+	if ! [ -x "$(command -v jq)" ]; then
+		echo "ERROR: Requirement not satisfied - jq -"
+		exit 1
+	fi
+
+        if [[ REDUCE -eq 1 && !(-x "$(command -v rdfind)") ]]; then
```
Comment (Rycieos) [inline factorio-map-process.sh:40]: If you take the above, this would probably need to change to:
```suggestion
        if [[ REDUCE -eq 1 ]] && ! command -v rdfind >/dev/null; then
```

File: factorio-map-process.sh [hunk 32]
```diff
@@ -0,0 +1,151 @@
...
+
+}
+
+function parseFileName() {
+	NAME_NOEXT="${FILENAME%.*}"
+	echo "Target File: ${FILENAME}"
+
+	IFS='\n'
+	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< "${NAME_NOEXT}"))
+	unset IFS
```
Comment (Rycieos) [inline factorio-map-process.sh]: I recently learned that setting then unsetting `$IFS` isn't a good idea, since you could be unsetting it from somewhere else (maybe even the users shell if they source the script for some reason. Better to use a local:
```suggestion
	local IFS='\n'
	infos=($(awk -F'[_.]' '{print $1; print $2}' <<< "${NAME_NOEXT}"))
```

**Review by Rycieos — COMMENTED** [review]
I'm getting real picky, probably because I'm currently working Liquidprompt. Sorry. Just let me know if your sick of it and I can clean some stuff up myself.

File: factorio-map-process.sh (hunk 32 above)
Comment (NateMeyer) [reply factorio-map-process.sh]: Neat, good to know.

File: factorio-map-process.sh (hunk 31 above)
Comment (NateMeyer) [reply factorio-map-process.sh:40]: I think the `command` check was giving me issues before.  I'd want to test this one before including it.  Feel free to finish the pull request and change it on your own if you're confident in it.

**Comment by Rycieos** [comment]
I just needed motivation to work on this again, which starting a new Factorio map was it.

I merged this as is, then made two fixup commits myself: 30b0c02 and 6ba6be3. I didn't change the functionality at all, just a few syntax fixes, two small bugs with `awk` and `jq`, and a few formatting tweaks.

Sorry about the 10 month delay, but thanks for the PR! This greatly cuts down the work of adding a new snapshot.
